packages/blockly/src/components/custom-block-form.tsx

```diff
@@ -225,76 +307,283 @@ export const CustomBlockForm: React.FC<IProps> = ({ blockType, editingBlock, exi
+                        onClick={() => setParameters(prev => prev.map((pp, idx) => idx===i ? ({ ...(pp as any), options: [ ...((pp as any).options||[]), ["", ""] ] }) as IParameter : pp))}
+                      >
+                        Add Option
+                      </button>
+                    </div>
+                  ) : (
+                    <div className={css.numberOptions} data-testid={`param-number-options-${i}`}>
+                      <div className={css.fieldGroup}>
+                        <label htmlFor={`default-value-${i}`}>Default Value</label>
+                        <input
+                          data-testid={`param-defaultValue-${i}`}
+                          id={`default-value-${i}`}
+                          placeholder="0"
+                          type="number"
+                          value={(p as any).defaultValue ?? ""}
+                          onChange={(e) => setParameters(prev => prev.map((pp, idx) => idx===i ? ({ ...(pp as any), defaultValue: Number(e.target.value) }) as IParameter : pp))}
+                        />
+                      </div>
+                    </div>
+                  )}
+                </div>
+              </div>
+            ))}
+          </div>
+        </div>
+      )}
+
+      {(formData.type === "creator" || formData.type === "setter") && (
+        <div className={css.customBlockForm_options} data-testid="section-options">
+          <label htmlFor="options">Options</label>
+          <div id="options">
+            {formData.options.map((option, index) => (
+              <div key={index} className={css.optionRow} data-testid={`option-row-${index}`}>
+                <input
+                  data-testid={`option-label-${index}`}
+                  placeholder="Display text (e.g., blue)"
+                  type="text"
+                  value={option.label}
+                  onChange={(e) => setFormData(prev => ({
+                    ...prev,
+                    options: prev.options.map((opt, i) => i === index ? { ...opt, label: e.target.value } : opt)
+                  }))}
+                />
+                <input
+                  data-testid={`option-value-${index}`}
+                  placeholder="Value (e.g., BLUE)"
+                  type="text"
+                  value={option.value}
+                  onChange={(e) => setFormData(prev => ({
+                    ...prev,
+                    options: prev.options.map((opt, i) => i === index ? { ...opt, value: e.target.value } : opt)
+                  }))}
+                />
+                <button
+                  data-testid={`remove-option-${index}`}
+                  type="button"
+                  onClick={() => setFormData(prev => ({
+                    ...prev,
+                    options: prev.options.filter((_, i) => i !== index)
+                  }))}
+                >
+                  Remove
+                </button>
+              </div>
```

Comment:
This is very similar to lines 421-444. Is there any way to abstract it into a separate reusable component? It's always good to look for opportunities like that, and I think this file has many of them.